src/components/KTextArea/KTextArea.vue

```diff
@@ -277,15 +312,44 @@ $kTextAreaPaddingY: var(--kui-space-40, $kui-space-40); // corresponds to mixin,
     }
   }
 
+  .input-wrapper {
+    display: contents;
+
+    &.autosize.legacy {
+      display: grid;
+
+      &::after {
+        content: attr(data-value) " ";
+        visibility: hidden;
+        white-space: pre-wrap;
+      }
+
+      .input-textarea {
+        overflow: hidden;
+      }
+    }
+
+    &.autosize {
+      .input-textarea {
+        field-sizing: content;
+        min-height: calc(($kTextAreaLineHeight * v-bind('rows')) + ($kTextAreaPaddingY * 2));
+      }
+    }
+  }
+
+  .input-wrapper.legacy::after,
   .input-textarea {
     @include inputDefaults;
 
-    // fixing mixed-decls deprecation: https://sass-lang.com/d/mixed-decls
     // stylelint-disable-next-line no-duplicate-selectors
     & {
-      min-height: calc(($kTextAreaLineHeight * 2) + ($kTextAreaPaddingY * 2)); // 2 lines + padding
-      resize: none;
+      grid-area: 1 / 1 / 2 / 2;
```

Comment:
question: why are we suddenly using grid properties?

Reply:
^ Same as above